libcudacxx/include/cuda/__vector_types/factory_functions.h

```diff
@@ -0,0 +1,844 @@
+{
+  return float4{};
+}
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr float4 make_float4(float __x, float __y, float __z, float __w) noexcept
+{
+  return float4{__x, __y, __z, __w};
+}
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr float4 make_float4(value_broadcast_t, float __x) noexcept
+{
+  return float4{__x, __x, __x, __x};
+}
+
+// make_doubleN
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr double1 make_double1() noexcept
+{
+  return double1{};
+}
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr double1 make_double1(double __x) noexcept
+{
+  return double1{__x};
+}
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr double1 make_double1(value_broadcast_t, double __x) noexcept
+{
+  return double1{__x};
+}
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr double2 make_double2() noexcept
+{
+  return double2{};
+}
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr double2 make_double2(double __x, double __y) noexcept
+{
+  return double2{__x, __y};
+}
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr double2 make_double2(value_broadcast_t, double __x) noexcept
+{
+  return double2{__x, __x};
+}
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr double3 make_double3() noexcept
+{
+  return double3{};
+}
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr double3 make_double3(double __x, double __y, double __z) noexcept
+{
+  return double3{__x, __y, __z};
+}
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr double3 make_double3(value_broadcast_t, double __x) noexcept
+{
+  return double3{__x, __x, __x};
+}
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr double4 make_double4() noexcept
+{
+  return double4{};
+}
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr double4
+make_double4(double __x, double __y, double __z, double __w) noexcept
+{
+  return double4{__x, __y, __z, __w};
+}
+
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr double4 make_double4(value_broadcast_t, double __x) noexcept
+{
+  return double4{__x, __x, __x, __x};
+}
+
+// make_vector
+
+template <class _Tp, size_t _Size>
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr vector_type_t<_Tp, _Size> make_vector() noexcept
+{
+  return vector_type_t<_Tp, _Size>{};
+}
+
+template <class _Tp, size_t _Size, class... _Args>
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr vector_type_t<_Tp, _Size> make_vector(_Args... __args) noexcept
+{
+  static_assert(sizeof...(_Args) == _Size, "number of arguments must match vector size");
+  return vector_type_t<_Tp, _Size>{__args...};
+}
+
+template <class _Tp, size_t _Size>
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr vector_type_t<_Tp, _Size>
+make_vector(value_broadcast_t, _CUDA_VSTD::type_identity_t<_Tp> __x) noexcept
+{
+  vector_type_t<_Tp, _Size> __ret{};
+  __ret.x = __x;
+  if constexpr (_Size > 1)
+  {
+    __ret.y = __x;
+  }
+  if constexpr (_Size > 2)
+  {
+    __ret.z = __x;
+  }
+  if constexpr (_Size > 3)
+  {
+    __ret.w = __x;
+  }
+  return __ret;
```

Comment:
I would rather have something like
```suggestion
  if constexpr (_Size == 1)
  {
    return vector_type_t<_Tp, _Size>{__x};
  }
  else if constexpr (_Size == 2)
  {
    return vector_type_t<_Tp, _Size>{__x, __x};
  }
  else if constexpr (_Size == 3)
  {
    return vector_type_t<_Tp, _Size>{__x, __x, __x};
  }
  else if constexpr (_Size == 4)
  {
    return vector_type_t<_Tp, _Size>{__x, __x, __x, __x};
  }
  else 
  {
    static_assert(_CUDA_VSTD::always_false_v<_Tp>, "Invalid vector size");
    return vector_type_t<_Tp, _Size>{};
  }
```